src/compositionGuards.ts

```diff
@@ -0,0 +1,12 @@
+import {isString} from './primitiveGuards';
+import {isDate} from './instanceGuards';
+import {isDateString} from './parseGuards';
+import {thenGuard, orGuard} from './Combinators';
+
+export const isDateOrStringyDate = orGuard(
```

Comment:
if this is a public repo, not a fan of 'stringy', even thought that's the original code.
how about isDateOrStringDate . 
or isDateOrDateString

in fact, i wonder if you could just have an orGuard(isDateString, isDate);.  so many possibilities